src/context/UserSessionProvider.jsx

```diff
@@ -0,0 +1,34 @@
+import { createContext, useEffect, useState } from "react";
+import { AUTH_TOKEN, axiosInstance } from "../axios";
+
+export const UserSessionContext = createContext();
+export const UserSessionProvider = ({ children }) => {
+  const [isLoggedIn, setIsLoggedIn] = useState(false);
+  useEffect(() => {
+    const token = localStorage.getItem(AUTH_TOKEN);
+    setIsLoggedIn(!!token);
+  });
```

Comment:
You can pass an empty dependency array as the second argument to run the effect only after the first render, otherwise it will run after every render.

```suggestion
  }, []);
```


Also, in this case you could set the initial state from localStorage:

```
const [isLoggedIn, setIsLoggedIn] = useState(!!localStorage.getItem(AUTH_TOKEN));
```